### ISSUE TITLE
Imagemagick Library Issue when trying to run Experhash

### ISSUE THREAD

**Issue by Faolain** [kemonomachi/experhash#5]
Hey I'm not really sure where to ask this, or if this is the appropriate place but figured you may have the most experience to address this or at least point me in the right direction.

I installed Imagemagick(7) through brew on MacOS Sierra (10.12) with seemingly no issue. I'm able to run `magick` in the command line etc. However when I try to run Experhash I receive the following error:

> dyld: Library not loaded: /usr/local/lib/libMagick++-6.Q16.5.dylib
> Referenced from: /Users/plato/Desktop/test_app/api/_build/test/lib/experhash/priv/experhash_port
> Reason: image not found

When I go and check the `/usr/local/lib/`  directory I see the following files exist:

> libMagick++-7.Q16HDRI.4.dylib
> libMagick++-7.Q16HDRI.dylib

but not `libMagick++-6.Q16.5.dylib` which I assume makes sense since I installed 7. Now I guess the confusion I have is that it seems the Experhash results are different than running Imagemagick directly from the command line which leads me to believe it may have to do with the makefile (https://github.com/kemonomachi/experhash/blob/master/Makefile#L10).

If it helps at all when I run,
`Magick++-config --cppflags --cxxflags --ldflags --libs` 

in the command line I get the following back:
```
-DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/usr/local/Cellar/imagemagick/7.0.7-30/include/ImageMagick-7
-DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/usr/local/Cellar/imagemagick/7.0.7-30/include/ImageMagick-7
-L/usr/local/Cellar/imagemagick/7.0.7-30/lib -lMagick++-7.Q16HDRI -lMagickWand-7.Q16HDRI -lMagickCore-7.Q16HDRI
-L/usr/local/Cellar/imagemagick/7.0.7-30/lib -lMagick++-7.Q16HDRI -lMagickWand-7.Q16HDRI -lMagickCore-7.Q16HDRI
```

And if it helps here are my environment variables within ~/.zshrc.

> export PATH="/usr/local/bin:$PATH"
> export MAGICK_HOME=“/usr/local/Cellar/imagemagick/7.0.7-30”
> export PATH="$MAGICK_HOME/bin:$PATH"
> export DYLD_LIBRARY_PATH="$MAGICK_HOME/lib/"
> export PKG_CONFIG_PATH="$MAGICK_HOME/lib/pkgconfig"

**Tl;DR**
It seems like the Makefile is referencing the wrong dylib and I'm not sure how to make it point to the "right" Imagemagick libraries which work when called straight from the command line.

**Comment by Faolain** [comment]
I found out what the problem was. I had an old version of Imagemagick installed which I guess compiled with experhash when I ran 

`mix deps.get`

I found the `hash.d` file within the deps/experhash/cpp/src directory and saw references to an older version of Imagemagick. After I cleared my deps and re-ran mix deps.get I saw that the file properly compiled with the new references to Imagemagick. Hopefully this helps someone else who comes across this.

**Comment by jpinnix** [comment]
@Faolain Thank you!!!
